bin/fathom-generate-events/eco-router-platoforms.ts

```diff
@@ -0,0 +1,33 @@
+import { BaseRoutablePlatform, RoutablePlatform, UniswapV2RoutablePlatform } from '@swapr/sdk'
+
+import { getChainNameByChainId } from './chain'
+
+interface PlatformInformation {
+  platformName: string
+  networkName: string
+  networkId: string
+}
+
+export function getMapOfExchanges(arg: any): PlatformInformation[] {
+  const isPrototypeOf = Function.call.bind(Object.prototype.isPrototypeOf)
+
+  if (!(isPrototypeOf(BaseRoutablePlatform, arg) || arg === BaseRoutablePlatform)) return []
```

Comment:
what does this condition mean? 

It would be cool if we put this condition in a variable and give it a descriptive name

Reply:
`Object.prototype.isPrototypeOf` is the equivalent of `instanceof` operator for classes. The condition is true when:

- current class is NOT a child of `BaseRoutablePlatform` (this the main abstract class that all EcoRouter platform extend)
OR
- the passed class is the actual `BaseRoutablePlatform` class.